OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ChildController.cs

```diff
@@ -101,6 +108,59 @@ public async Task<IActionResult> GetUsersChildById(Guid id)
             return Ok(await service.GetByIdAndUserId(id, userId).ConfigureAwait(false));
         }
 
+        /// <summary>
+        /// Get approved children from the database by workshop id.
+        /// </summary>
+        /// <param name="workshopId">Id of the parent.</param>
+        /// <param name="offsetFilter">Filter to get a part of all children that were found.</param>
+        /// <returns>The result is a <see cref="SearchResult{ChildDto}"/> that contains the count of all found children and a list of children that were received.</returns>
+        [HasPermission(Permissions.WorkshopEdit)]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<ChildDto>))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [HttpGet("{workshopId}")]
+        public async Task<IActionResult> GetApprovedByWorkshopId(Guid workshopId, [FromQuery] OffsetFilter offsetFilter)
+        {
+            var userHasRights = await this.IsUserProvidersOwnerOrAdmin(workshopId).ConfigureAwait(false);
+            if (!userHasRights)
+            {
+                return StatusCode(403, "Forbidden for another providers.");
+            }
+
+            return Ok(await service.GetApprovedByWorkshopId(workshopId, offsetFilter).ConfigureAwait(false));
+        }
+
+        private async Task<bool> IsUserProvidersOwnerOrAdmin(Guid workshopId)
+        {
+            Guid providerId = await providerService.GetProviderIdForWorkshopById(workshopId);
+
+            if (User.IsInRole(nameof(Role.Provider).ToLower()))
+            {
+                var userId = User.FindFirst("sub")?.Value;
```

Comment:
"sub" - is a string constant, that is probably used not only here. Please extract it or surrounding code (i.e. `User.FindFirst("sub")?.Value`) into some "entity" (const/helper method/extension method/your choice).

Reply:
We have a module for it: OutOfSchool.WebApi.Common.GettingUserProperties